src/client/csfle/client_encryption.rs

```diff
@@ -0,0 +1,354 @@
+//! Support for explicit encryption.
```

Comment:
This is perhaps more appropriate for a later PR, but I think it would be beneficial to have more extended documentation of what this module is for, how it differs from auto encryption, and some doc-examples to demonstrate basic usage. Ditto for adding examples to any of the new types as you think is worth it.